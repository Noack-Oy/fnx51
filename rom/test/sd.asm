
; ************************
; * SD Card Test Program *
; ************************

; This program initializes an SD card
; and dumps the first sector.


; **********************
; * Header Definitions *
; **********************
.inc ../global/variables.equ
.inc ../global/sfr.equ
.inc ../serial/sfr.equ
.inc ../spi/sfr.equ
.inc ../sd/config.equ

; *************
; * Main Code *
; *************

; *** test/sd.asm ***

	.org	0
.inc	../global/init.inc

	acall	serial_init
	acall	sd_init
	acall	print_hex_8	; status
	acall	print_hex_32	; operation condition
	mov	a,#13
	acall	print_char
	mov	a,#10
	acall	print_char

	; use xram
	anl	auxr,#0xe1	; clear extram, xrs0-2
	orl	auxr,#0x10	; set xrs2 (size 1792 bytes)

	clr	a
	mov	dpl,a
	mov	dph,a
	mov	r0,a
	mov	r1,a
	mov	r2,a
	mov	r3,a
	acall	sd_block_read

	; print CRC
	mov	a,r4
	mov	r0,a
	mov	a,r5
	mov	r1,a
	acall	print_hex_16
	mov	a,#13
	acall	print_char
	mov	a,#10
	acall	print_char

	clr	a
	mov	r0,a
	mov	r1,a
	mov	r2,a
	mov	r3,a
	mov	stream_in,a
	mov	stream_in+1,a
	mov	dptr,#stream_xram_read
	mov	in,dpl
	mov	in+1,dph
	acall	dump
	acall	dump

halt:
	sjmp	halt

; *********************
; * Library Functions *
; *********************
.inc ../serial/init.inc
.inc ../serial/tx.inc
.inc ../serial/rx.inc

.inc ../spi/transfer.inc
.inc ../sd/init.inc
.inc ../sd/warmup.inc
.inc ../sd/select.inc
.inc ../sd/cmd0.inc
.inc ../sd/cmd8.inc
.inc ../sd/acmd41.inc
.inc ../sd/cmd55.inc
.inc ../sd/cmd58.inc
.inc ../sd/command.inc
.inc ../sd/response.inc
.inc ../sd/block_read.inc

.inc ../util/delay.inc
.inc ../util/regbank.inc
.inc ../print/char.inc
.inc ../print/hex.inc
.inc ../print/text.inc
.inc ../stream/xram_read.inc
.inc ../read/char.inc
.inc ../util/xch.inc
.inc ../util/dump.inc

.equ panic_out, serial_tx
.inc ../util/panic.inc
