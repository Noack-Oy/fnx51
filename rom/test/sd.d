sd.lst sd.hex: sd.asm ../global/variables.equ ../global/sfr.equ ../serial/sfr.equ ../spi/sfr.equ ../sd/config.equ ../global/init.inc ../serial/init.inc ../serial/tx.inc ../serial/rx.inc ../spi/transfer.inc ../sd/init.inc ../sd/warmup.inc ../sd/select.inc ../sd/cmd0.inc ../sd/cmd8.inc ../sd/acmd41.inc ../sd/cmd55.inc ../sd/cmd58.inc ../sd/command.inc ../sd/response.inc ../sd/block_read.inc ../util/delay.inc ../util/regbank.inc ../print/char.inc ../print/hex.inc ../print/text.inc ../stream/xram_read.inc ../read/char.inc ../util/xch.inc ../util/dump.inc ../util/panic.inc
