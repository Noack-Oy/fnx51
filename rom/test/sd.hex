:1000000075815090006E85823085833190006685C1
:100010008232858333115C118151145134740D5136
:100020000F740A510F538EE1438E10E4F582F5836D
:10003000F8F9FAFB3190ECF8EDF95129740D510FF4
:10004000740A510FE4F8F9FAFBF534F53590025BC8
:100050008582308583315190519080FE759852751C
:100060009ABF759B1E223099FDC299F5992230984E
:10007000FDC298E59922F5C5E5C4548060FAE5C54E
:1000800022EDC0E075C3F111BE11D37D0511E2B4BC
:1000900001281D11F2B401221DBB001EBA001BB9BC
:1000A0000118B8AA157D64310A600A740A31C7DDE7
:1000B000F67D0280057D0131321DD0E0CD22C0E009
:1000C000C0F0D29175F00A74FF1176D5F0F9D0F036
:1000D000D0E022C0E074FF1176C29174FF1176D097
:1000E000E02231E3E4F8F9FAFB7C943146316031E7
:1000F000FC22ECC0E0740878AA79017A007B007CCD
:100100008631463172FCD0E0CC223121B40111316C
:10011000E3E4F8F9FA7B40FC74293146316031FCA4
:100120002231E3E4F8F9FAFBFC74373146316031EF
:10013000FC22ECC0E0E4F8F9FAFBFC743A314631F9
:1001400072FCD0E0CC22C0E044401176EB1176EA9C
:100150001176E91176E81176EC44011176D0E022AF
:10016000C0F08508F074FF1176B4FF03D5F0F6D027
:10017000F0223160B40200400122F874FF1176FBD6
:1001800074FF1176FA74FF1176F974FF1176C822A4
:10019000C0E074117CFF31463160B4FEFB7D007419
:1001A000FF1176F0A374FF1176F0A3DDF274FF1156
:1001B00076FD74FF1176FC740128F8E439F9E43A0D
:1001C000FAE43BFBD0E022C0E0C0F075F0B9C0E03B
:1001D000C0F0A4A4D0F0D0E0D5F0F3D5E0EDD0F09D
:1001E000D0E022C0E0E5D05418B41802714CE5D03C
:1001F0002408F5D0541824F8F8D0E022C0E0E5D067
:1002000054187002714CE5D024F8F5D0D0E022C02B
:1002100032C03322C45118C4C0E0540F2490D434E7
:1002200040D44420510FD0E022C0E0E95114E851FD
:1002300014D0E022C0E0EB5114EA5114E95114E863
:100240005114D0E022C0E0E8C0E0E4F89360065129
:100250000F08E870F7D0E0F8D0E022C08EC082C06E
:1002600083E58E55E14510F58E853482853583E032
:10027000A3858234858335D083D082D08E22C0304E
:10028000C03122C8CCC8C9CDC9CACECACBCFCB22B7
:10029000C0F0C0E0F5F0ECC0E0EDC0E0EEC0E0EF93
:1002A000C0E05183E581F82410F5817910517E0872
:1002B000F6D5F003198002D9F45183513451837477
:1002C0003A510FE824F029F87A1008EA5403700430
:1002D0007420510FEAD39950087420510F510F80A8
:1002E00003E651147420510FDAE0E824F0F874208A
:1002F000510F747C510F7410C399F9FA08E6B420B9
:10030000004005B47F004002742E510FDAEE747C79
:10031000510F740D510F740A510FE824F0F8EC29B5
:10032000FCED3400FDEE3400FEEF3400FFE5F0603C
:100330000241ABE58124F0F5815183D0E0FFD0E0AC
:10034000FED0E0FDD0E0FCD0E0D0F022C2AF9000C3
:10035000668582328583339003665145D0E051141F
:10036000D0E0511480FE0D0A202121212070616E01
:0503700069632040005C
:00000001FF
