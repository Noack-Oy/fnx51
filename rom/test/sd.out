BOOTLOADER
> U
< U
> :020000050B00EE
< :020000050B00EE7B.\r\n
WRITE: sd.hex
> :1000000075815090006E85823085833190006685C1
< :1000000075815090006E85823085833190006685C1.\r\n
> :100010008232858333115C118151145134740D5136
< :100010008232858333115C118151145134740D5136.\r\n
> :100020000F740A510F538EE1438E10E4F582F5836D
< :100020000F740A510F538EE1438E10E4F582F5836D.\r\n
> :10003000F8F9FAFB3190ECF8EDF95129740D510FF4
< :10003000F8F9FAFB3190ECF8EDF95129740D510FF4.\r\n
> :10004000740A510FE4F8F9FAFBF534F53590025BC8
< :10004000740A510FE4F8F9FAFBF534F53590025BC8.\r\n
> :100050008582308583315190519080FE759852751C
< :100050008582308583315190519080FE759852751C.\r\n
> :100060009ABF759B1E223099FDC299F5992230984E
< :100060009ABF759B1E223099FDC299F5992230984E.\r\n
> :10007000FDC298E59922F5C5E5C4548060FAE5C54E
< :10007000FDC298E59922F5C5E5C4548060FAE5C54E.\r\n
> :1000800022EDC0E075C3F111BE11D37D0511E2B4BC
< :1000800022EDC0E075C3F111BE11D37D0511E2B4BC.\r\n
> :1000900001281D11F2B401221DBB001EBA001BB9BC
< :1000900001281D11F2B401221DBB001EBA001BB9BC.\r\n
> :1000A0000118B8AA157D64310A600A740A31C7DDE7
< :1000A0000118B8AA157D64310A600A740A31C7DDE7.\r\n
> :1000B000F67D0280057D0131321DD0E0CD22C0E009
< :1000B000F67D0280057D0131321DD0E0CD22C0E009.\r\n
> :1000C000C0F0D29175F00A74FF1176D5F0F9D0F036
< :1000C000C0F0D29175F00A74FF1176D5F0F9D0F036.\r\n
> :1000D000D0E022C0E074FF1176C29174FF1176D097
< :1000D000D0E022C0E074FF1176C29174FF1176D097.\r\n
> :1000E000E02231E3E4F8F9FAFB7C943146316031E7
< :1000E000E02231E3E4F8F9FAFB7C943146316031E7.\r\n
> :1000F000FC22ECC0E0740878AA79017A007B007CCD
< :1000F000FC22ECC0E0740878AA79017A007B007CCD.\r\n
> :100100008631463172FCD0E0CC223121B40111316C
< :100100008631463172FCD0E0CC223121B40111316C.\r\n
> :10011000E3E4F8F9FA7B40FC74293146316031FCA4
< :10011000E3E4F8F9FA7B40FC74293146316031FCA4.\r\n
> :100120002231E3E4F8F9FAFBFC74373146316031EF
< :100120002231E3E4F8F9FAFBFC74373146316031EF.\r\n
> :10013000FC22ECC0E0E4F8F9FAFBFC743A314631F9
< :10013000FC22ECC0E0E4F8F9FAFBFC743A314631F9.\r\n
> :1001400072FCD0E0CC22C0E044401176EB1176EA9C
< :1001400072FCD0E0CC22C0E044401176EB1176EA9C.\r\n
> :100150001176E91176E81176EC44011176D0E022AF
< :100150001176E91176E81176EC44011176D0E022AF.\r\n
> :10016000C0F08508F074FF1176B4FF03D5F0F6D027
< :10016000C0F08508F074FF1176B4FF03D5F0F6D027.\r\n
> :10017000F0223160B40200400122F874FF1176FBD6
< :10017000F0223160B40200400122F874FF1176FBD6.\r\n
> :1001800074FF1176FA74FF1176F974FF1176C822A4
< :1001800074FF1176FA74FF1176F974FF1176C822A4.\r\n
> :10019000C0E074117CFF31463160B4FEFB7D007419
< :10019000C0E074117CFF31463160B4FEFB7D007419.\r\n
> :1001A000FF1176F0A374FF1176F0A3DDF274FF1156
< :1001A000FF1176F0A374FF1176F0A3DDF274FF1156.\r\n
> :1001B00076FD74FF1176FC740128F8E439F9E43A0D
< :1001B00076FD74FF1176FC740128F8E439F9E43A0D.\r\n
> :1001C000FAE43BFBD0E022C0E0C0F075F0B9C0E03B
< :1001C000FAE43BFBD0E022C0E0C0F075F0B9C0E03B.\r\n
> :1001D000C0F0A4A4D0F0D0E0D5F0F3D5E0EDD0F09D
< :1001D000C0F0A4A4D0F0D0E0D5F0F3D5E0EDD0F09D.\r\n
> :1001E000D0E022C0E0E5D05418B41802714CE5D03C
< :1001E000D0E022C0E0E5D05418B41802714CE5D03C.\r\n
> :1001F0002408F5D0541824F8F8D0E022C0E0E5D067
< :1001F0002408F5D0541824F8F8D0E022C0E0E5D067.\r\n
> :1002000054187002714CE5D024F8F5D0D0E022C02B
< :1002000054187002714CE5D024F8F5D0D0E022C02B.\r\n
> :1002100032C03322C45118C4C0E0540F2490D434E7
< :1002100032C03322C45118C4C0E0540F2490D434E7.\r\n
> :1002200040D44420510FD0E022C0E0E95114E851FD
< :1002200040D44420510FD0E022C0E0E95114E851FD.\r\n
> :1002300014D0E022C0E0EB5114EA5114E95114E863
< :1002300014D0E022C0E0EB5114EA5114E95114E863.\r\n
> :100240005114D0E022C0E0E8C0E0E4F89360065129
< :100240005114D0E022C0E0E8C0E0E4F89360065129.\r\n
> :100250000F08E870F7D0E0F8D0E022C08EC082C06E
< :100250000F08E870F7D0E0F8D0E022C08EC082C06E.\r\n
> :1002600083E58E55E14510F58E853482853583E032
< :1002600083E58E55E14510F58E853482853583E032.\r\n
> :10027000A3858234858335D083D082D08E22C0304E
< :10027000A3858234858335D083D082D08E22C0304E.\r\n
> :10028000C03122C8CCC8C9CDC9CACECACBCFCB22B7
< :10028000C03122C8CCC8C9CDC9CACECACBCFCB22B7.\r\n
> :10029000C0F0C0E0F5F0ECC0E0EDC0E0EEC0E0EF93
< :10029000C0F0C0E0F5F0ECC0E0EDC0E0EEC0E0EF93.\r\n
> :1002A000C0E05183E581F82410F5817910517E0872
< :1002A000C0E05183E581F82410F5817910517E0872.\r\n
> :1002B000F6D5F003198002D9F45183513451837477
< :1002B000F6D5F003198002D9F45183513451837477.\r\n
> :1002C0003A510FE824F029F87A1008EA5403700430
< :1002C0003A510FE824F029F87A1008EA5403700430.\r\n
> :1002D0007420510FEAD39950087420510F510F80A8
< :1002D0007420510FEAD39950087420510F510F80A8.\r\n
> :1002E00003E651147420510FDAE0E824F0F874208A
< :1002E00003E651147420510FDAE0E824F0F874208A.\r\n
> :1002F000510F747C510F7410C399F9FA08E6B420B9
< :1002F000510F747C510F7410C399F9FA08E6B420B9.\r\n
> :10030000004005B47F004002742E510FDAEE747C79
< :10030000004005B47F004002742E510FDAEE747C79.\r\n
> :10031000510F740D510F740A510FE824F0F8EC29B5
< :10031000510F740D510F740A510FE824F0F8EC29B5.\r\n
> :10032000FCED3400FDEE3400FEEF3400FFE5F0603C
< :10032000FCED3400FDEE3400FEEF3400FFE5F0603C.\r\n
> :100330000241ABE58124F0F5815183D0E0FFD0E0AC
< :100330000241ABE58124F0F5815183D0E0FFD0E0AC.\r\n
> :10034000FED0E0FDD0E0FCD0E0D0F022C2AF9000C3
< :10034000FED0E0FDD0E0FCD0E0D0F022C2AF9000C3.\r\n
> :10035000668582328583339003665145D0E051141F
< :10035000668582328583339003665145D0E051141F.\r\n
> :10036000D0E0511480FE0D0A202121212070616E01
< :10036000D0E0511480FE0D0A202121212070616E01.\r\n
> :0503700069632040005C
< :0503700069632040005C.\r\n
> :00000001FF
< :00000001FF.\r\n
RESET
00c0ff8000
8987
00000000: fa b8 00 10  8e d0 bc 00  b0 b8 00 00  8e d8 8e c0  |................|
00000010: fb be 00 7c  bf 00 06 b9  00 02 f3 a4  ea 21 06 00  |...|.........!..|
00000020: 00 be be 07  38 04 75 0b  83 c6 10 81  fe fe 07 75  |....8.u........u|
00000030: f3 eb 16 b4  02 b0 01 bb  00 7c b2 80  8a 74 01 8b  |.........|...t..|
00000040: 4c 02 cd 13  ea 00 7c 00  00 eb fe 00  00 00 00 00  |L.....|.........|
00000050: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000060: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000070: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000080: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000090: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000a0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000b0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000c0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000d0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000e0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000f0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000100: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000110: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000120: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000130: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000140: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000150: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000160: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000170: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000180: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000190: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000001a0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000001b0: 00 00 00 00  00 00 00 00  54 b3 4d 35  00 00 00 02  |........T.M5....|
000001c0: 03 00 0c fe  7f e8 80 00  00 00 00 e8  77 00 00 00  |............w...|
000001d0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000001e0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000001f0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 55 aa  |..............U.|
